src/structstore_shared.hpp

```diff
@@ -42,62 +44,192 @@ class StructStoreShared {
         ~SharedData() = delete;
     };
 
-    int shm_fd;
     std::string shm_path;
+    int shm_fd;
     SharedData* shm_ptr;
+    size_t bufsize;
+    bool create;
+    bool reinit;
 
 public:
-    explicit StructStoreShared(const std::string& shm_path, size_t bufsize = 2048) : shm_path(shm_path) {
-        shm_fd = shm_open(shm_path.c_str(), O_CREAT | O_RDWR, 0600);
-        size_t size = sizeof(SharedData) + bufsize;
 
-        // check if shared memory already exists
+    explicit StructStoreShared(
+            const std::string& shm_path,
+            size_t bufsize = 2048,
+            bool create = true,
+            bool reinit = false)
```

Comment:
See comment above. This flag means `invalidate existing stores before creation` and `invalidate store after destruction`, which is not clear by this parameter name, maybe `reinit_and_cleanup` instead?